.github/workflows/check_cla_assigned.yml

```
@@ -0,0 +1,176 @@
+      pull-requests: write
+      issues: write
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16'
+      - name: Install Dependencies
+        run: npm install googleapis google-auth-library
+      - name: Check CLA signature
+        id: check-cla
+        uses: actions/github-script@v6
+        env:
+          SHEETS_CRED: '${{ secrets.SHEETS_CRED }}'
+          SPREADSHEET_ID: '${{ secrets.SPREADSHEET_ID }}'
+          USERNAME: '${{ github.event.pull_request.user.login || github.event.assignee.login }}'
+        with:
+          script: |
+            try {
+              const sheetsCredRaw = process.env.SHEETS_CRED;
+              let sheetsCredJson;
+              try {
+                sheetsCredJson = JSON.parse(sheetsCredRaw);
+                console.log("SHEETS_CRED parsed successfully");
+              } catch (parseError) {
+                console.error("Failed to parse SHEETS_CRED:", parseError.message);
+                console.log("First 10 chars:", sheetsCredRaw.substring(0, 10) + "...");
+                throw parseError;
+              }
+              console.log("SHEETS_CRED loaded successfully");
+              const username = process.env.USERNAME;
+              console.log(`Checking CLA signature for user: ${username}`);
+
+              const { google } = require('googleapis');
+              const { JWT } = require('google-auth-library');
+              const { OAuth2Client } = require('google-auth-library');
+
+              if (!sheetsCredJson.installed || !sheetsCredJson.installed.client_id || !sheetsCredJson.installed.client_secret || !sheetsCredJson.refresh_token) {
+                console.error("SHEETS_CRED is missing required credential fields");
+                core.warning("SHEETS_CRED is missing required credential fields");
+                core.setOutput('has-signed-cla', 'true');
```

is this "fail open" policy desired? CLA signing has legal implications for Oppia, so we might want to fail closed (i.e. default to blocking PRs if we can't confirm that the CLA has been signed)

btw it looks like oppiabot also fails open

/cc @seanlip

---

@U8NWXD Confirming that not "failing open" is the correct thing to do, i.e. if we cannot confirm signing then this check should fail, not pass. Thanks for catching this!

Also, if there is some way to convey "not true, not false, there's some error in the process" (i.e. "unable to verify") then that would probably be semantically better here than just saying "false". In one case we can say "an error occurred, please contact an admin" and in the other case we can say "you haven't signed it, please do so". In fact in the latter case it is fine to close their PR and ask them to reopen once they've signed, but we wouldn't want to do that in the former.

---

That's what I wanna discuss, thanks for raising it. Marked false.

---

How about we keep the same `true` vs `false` output values, but if we can't tell whether the CLA has been signed, we throw an error and fail the workflow? That will block PR merges (so long as a passing workflow is required for PR merging) but not close the PR. It also takes advantage of GitHub's built-in error handling rather than building something new, and throwing an error correctly conveys that something went wrong

---

Hmm, that's great way to follow. Now I'm logging the error and raising up the warning over the step and hence throwing error to fail the workflow. PTAL there.